pymeasure/instruments/keithley/keithley2281S.py

```diff
@@ -0,0 +1,573 @@
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import enum
+import time
+
+import numpy as np
+import pandas as pd
+
+from pymeasure.instruments import Instrument, SCPIMixin
+from pymeasure.instruments.channel import Channel
+from pymeasure.instruments.keithley.buffer import KeithleyBuffer
+from pymeasure.instruments.validators import strict_discrete_set, strict_range, truncated_range
+
+
+@enum.unique
+class Keithley2281SOperationEventRegister(enum.IntFlag):
+    """
+    Enum containing Keithley2281S Operation Instrument Summary Event Register definition
+    """
+
+    CALIBRATION = 1  # Performing calibration
+    _RESERVED_1 = 2
+    _RESERVED_2 = 4
+    _RESERVED_3 = 8
+    MEASUREMENT = 16  # Performing measurement
+    TRIGGER = 32  # Waiting for trigger
+    ARM = 64  # Waiting for arm event
+    _RESERVED_4 = 128
+    FILTER_SETTLED = 256  # Filter has settled or is disabled
+    LIST = 512  # Running a list
+    IDLE = 1024  # Idling
+    _RESERVED_5 = 2048
+    _RESERVED_6 = 4096
+    _RESERVED_7 = 8192
+    _RESERVED_8 = 16384
+
+
+@enum.unique
+class Keithley2281SMeasurementEventRegister(enum.IntFlag):
+    """
+    Enum containing Keithley2281S Measurement Instrument Summary Event Register definition
+    """
+
+    READING_OVERFLOW = 1
+    _RESERVED_1 = 2
+    _RESERVED_2 = 4
+    _RESERVED_3 = 8
+    _RESERVED_4 = 16
+    _RESERVED_5 = 32
+    READING_AVAILABLE = 64
+    BUFFER_HALF_FULL = 128
+    BUFFER_FULL = 256
+    BUFFER_THREE_QUARTERS_FULL = 512
+    BUFFER_ONE_QUARTER_FULL = 1024
+    _RESERVED_6 = 2048
+    _RESERVED_7 = 4096
+    _RESERVED_8 = 8192
+    _RESERVED_9 = 16384
+
+
+# Constants needed in multiple classes
+_VOLTAGE_RANGE = [0.0, 20]
+_CURRENT_RANGE_BT_BS = [0.0, 6.1]
+_INTERNAL_MEMORY_SLOTS = list(range(1, 10))
+_PLC_RANGE = [0.002, 12]
+_CURRENT_RANGE = [0.1, 6.1]
+
+
+class PowerSupplyChannel(Channel):
+    """
+    Power Supply Mode of the device
+
+    Make sure that the parent's instrument `function_mode` property is set to `POWER` before using
+    this channel.
+    """
+
+    _query_buffer_data = Instrument.measurement(
+        ':DATA:DATA? "READ,SOUR,REL"',
+        """Get the buffer in power supply mode""",
+        separator=",",
+    )
```

Comment:
I'd change the name of the property. Right now it sounds like a method. Maybe something like `_data_buffer` is better.

Small note: Add a period at the end of the docstring.